thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/accessControl/AccessControlConfiguration.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2022 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package ai.startree.thirdeye.spi.accessControl;
+
+import java.util.Map;
+
+public class AccessControlConfiguration {
+  public boolean enabled;
```

Comment:
We may need to think a bit more about the config structure. this is a plugin config that we are trying to pass on here. am i right?

This is a rough take. but I think containerizing this might be beneficial in the long run.
```
# inside server.yaml

auth: # Auth Config
  enabled: true
  # other auth conf

  plugins: # AuthPluginConfigs <String, Map<>>
    startree-auth: # only this map gets pushed to the plugin.
      key1: value1
      key2: value2

```

Reply:
Ya, this format looks good. (tbh I was being a little lazy w/ design 😅 ) Instead of passing only the `startee-auth` field, I'll pass the whole `plugins` field and the plugin itself looks for the `startree-auth` field. This way the plugin loader doesn't have to know how to map config entries to individual plugins.